### ISSUE TITLE
Bad complex return values & documentation for linalg.lstsq (migrated from Trac #937)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2490]
Original ticket http://projects.scipy.org/numpy/ticket/937
Reported 2008-10-17 by atmention:nilswagner01, assigned to unknown.

I am confused by the output of lstsq in case of complex matrices.
residues (Euclidian norm) should be real.

{{{>>> numpy.**version**
'1.3.0.dev5945'
}}}

If I run the attached file I get

```
residues should be real but is complex [-0.04002398-0.01310083j]
residues is empty [] 0
```

**Comment by thouis** [comment]
Attachment in Trac by atmention:nilswagner01, 2008-10-17: [test_lstsq.py](http://projects.scipy.org/numpy/attachment/ticket/937/test_lstsq.py)

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2008-11-28

Hmm, the documentation is also incorrect. The squared norm is minimized, not the norm. And it should be the Euclidean norm, not just any norm. I think the sum of squares explanation is a bit clearer. Also rank should be computed rank, as this can differ from the "true" rank. Residues should be replaced by residuals, etc.

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2009-07-13

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2009-07-13

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2009-07-13

**Comment by thouis** [comment]
Comment in Trac by atmention:nilswagner01, 2010-01-16

Scipy linalg.lstsq is also broken for complex matrices
    from numpy.random import rand
    from numpy.linalg import lstsq
    from scipy.linalg import lstsq as lstsq_s
    A = rand(20,10)+1j_rand(20,10)
    b = rand(20)+1j_rand(20)
    x,res,rank,s = lstsq(A,b)
    x_s,res_s,rank_s, s_s = lstsq_s(A,b)

**Comment by thouis** [comment]
Comment in Trac by trac user samueljohn, 2010-08-16

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-10-10

Fixed in commit:99f79e3 commit:ab065f2 in Numpy, and revision 6832 in Scipy
